c_stubs/xenctrlext_stubs.c

```diff
@@ -435,6 +435,42 @@ CAMLprim value stub_xenctrlext_get_msr_arch_caps(value xch) {
 	CAMLreturn(caml_copy_int64(val));
 }
 
+CAMLprim value stub_xc_cpu_policy_get_system(value xch, value idx, value arr)
+{
+    CAMLparam3(xch, idx, arr);
+    int retval = xc_cpu_policy_get_system(_H(xch), Int_val(idx), &arr);
```

Comment:
I don't think &arr is correct here, you are passing the address of an ocaml int64 array directly to a C function. If you want to do that you need to use bigarrays instead of int64 array, or you need to copy the array.